cmake/Modules/ThirdpartyToolchain.cmake

```diff
@@ -123,6 +123,13 @@ if ("${CMAKE_RAY_LANG_PYTHON}" STREQUAL "YES")
       "PYARROW_WITH_PARQUET=1"
       "PYARROW_PARALLEL=")
 
+    if (APPLE)
+      # Since 10.14, the XCode toolchain only accepts libc++ as the
+      # standard library. This should also work on macOS starting from 10.9.
+      set(pyarrow_ENV ${pyarrow_ENV} "CXXFLAGS='-stdlib=libc++'")
+      set(pyarrow_ENV ${pyarrow_ENV} "MACOSX_DEPLOYMENT_TARGET=10.7")
```

Comment:
@pcmoritz I'm getting the following error, unless I change `10.7` to `10.14` here. Any idea why?
```
cmake -DPYTHON_EXECUTABLE=/Users/haochen/.pyenv/versions/3.6.5/bin/python  -DPYARROW_BUILD_PARQUET=on -DPYARROW_BOOST_USE_SHARED=on -DPYARROW_BUILD_PLASMA=on -DPYARROW_USE_TENSORFLOW=on -DPYARROW_BUNDLE_ARROW_CPP
=ON -DCMAKE_BUILD_TYPE=release /Users/haochen/code/ant_ray/build/external/arrow/src/arrow_ep/python
error: $MACOSX_DEPLOYMENT_TARGET mismatch: now "10.7" but "10.14" during configure
```
I'm on macOS 10.14.